src/server/db_slice.h

```diff
@@ -547,7 +547,7 @@ class DbSlice {
     ExpireIterator exp_it;
   };
 
-  PrimeItAndExp ExpireIfNeeded(const Context& cntx, PrimeIterator it) const;
+  PrimeItAndExp ExpireIfNeeded(const Context& cntx, PrimeIterator it, bool preempts = false) const;
```

Comment:
why do you have default , are there aditinal flows which I dont see where this function is called and I can not review if preemption is allowed or not in them

Reply:
so you are suggesting to disable default arguments to function ? I will push a change quickly, but what's the point of `default arguments` then -- to avoid passing explicitly the `false` argument.....

Reply:
Are you asking me if I want generally to disable default arguments to functions in cpp ? no 
But here we need to understand all the flows that dont allow preeption and later make sure we dont invoke this while big value serialization is in progress.
So actually if I would think to use a default I would choose the default true, cause by default we want all flows to write the journal but we have very specific flows that we want to protect that preemption is not allowed there

Reply:
:shrug: